source/common/router/retry_state_impl.h

```diff
@@ -55,10 +57,10 @@ class RetryStateImpl : public RetryState {
   Event::Dispatcher& dispatcher_;
   uint32_t retry_on_{};
   uint32_t retries_remaining_{1};
-  uint32_t current_retry_{};
   DoRetryCallback callback_;
   Event::TimerPtr retry_timer_;
   Upstream::ResourcePriority priority_;
+  BackOffStrategyPtr backoff_strategy_ptr_;
```

Comment:
FWIW, I'm not a fan of putting `_ptr` in the variable name unless it is really needed to make things clearer (e.g. to disambiguate). Here, I would just call this `backoff_strategy_`. I think you'll find this consistent with other Envoy code (e.g. `retry_timer_` above).